core/src/main/java/com/linecorp/armeria/common/auth/OAuth1aTokenBuilder.java

```diff
@@ -125,13 +137,60 @@ public OAuth1aTokenBuilder version(String version) {
     }
 
     /**
-     * Sets additional (or user-defined) parameters.
+     * Sets additional parameters. If the specified {@code additionals} has pre-defined parameters,
+     * the parameters are set automatically to this builder and removed from the {@code additionals}.
      */
     public OAuth1aTokenBuilder additionals(Map<String, String> additionals) {
-        this.additionals = ImmutableMap.copyOf(requireNonNull(additionals, "additionals"));
+        setParameters(this, requireNonNull(additionals, "additionals"));
         return this;
     }
 
+    private void additionals0(ImmutableMap<String, String> additionals) {
+        this.additionals = additionals;
+    }
```

Comment:
- Could this method be inlined?
- Could we merge instead of overwriting the previous additionals?